### DISCUSSION TITLE
Pretty basic but how can I press {esc} key over the page?

### DISCUSSION THREAD

**Discussion by MatiDiez** [cypress-io/cypress#17557] [Questions and Help] [answered]
Hey everyone, I'm testing a page. One of my steps enebles/disables a pop up in the page. 
Some times it takes a bit longer to disable the popup on the backend, so on my subsequent test, I randomly get or don't get the pop up. 

So, the easiest straight forward solution for me, is to press Escape key. If the pop up is present it will be closed, if it is not, nothing happens. So my question is: 

How to I simulate pressing the Escape key? 

I tried `cy.type('{esc}')` which obviously doesn't work because it has to be chained to a DOM element. I also tried 
`cy.get('body').trigger('keydown', { keyCode: 27})` but it didn't work. Any suggestions?

**Comment by msmosavar** [comment]
maybe too late but this worked for me:
`cy.get('body').type('{esc}')`

**Comment by jennifer-shehane** [comment, marked as answer]
As of Cypress 15.3.0 you can dispatch a real, native Escape keypress with:

```js
cy.press('Escape')
```

Unlike .type(), cy.press() fires the key at the browser level (not a synthetic DOM event) and doesn’t need to be chained off an element. 
On older versions, cy.get('body').type('{esc}') is still the way to go.
